### PR TITLE
Insert function documentation in editor

### PR DESCRIPTION
**Pull request by STetsing**
From now on, when the user right click on any function, then select `generate documentation`, the generated documentation in fitted above the function

The user can now chat with the terminal outputs

### CODE REVIEW

**Comment by STetsing** [comment]
@LianaHus what is the status of this pr?

**Comment by STetsing** [comment]
@LianaHus please review again

File: apps/remix-ide/src/app/plugins/solcoderAI.tsx [hunk 1]
```diff
@@ -47,6 +47,12 @@ export class SolCoder extends Plugin {
     this.solgpt_chat_history = []
   }
 
+  pushChatHistory(prompt, result){
+    const chat:ChatEntry = [prompt, result.data[0]]
+    this.solgpt_chat_history.push(chat)
+    if (this.solgpt_chat_history.length >this.max_history){this.solgpt_chat_history.shift()}
```
Comment (Aniket-Engg) [inline apps/remix-ide/src/app/plugins/solcoderAI.tsx:53]: there should be space after `>`, before `{` and wherever required to make it better readable

File: libs/remix-ui/editor/src/lib/providers/documentationProvider.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  constructor(completion: any){
+    this.completion = completion
+  }
+
+  async provideInlineCompletions(model: monacoTypes.editor.ITextModel, position: monacoTypes.Position, context: monacoTypes.languages.InlineCompletionContext, token: monacoTypes.CancellationToken): Promise<monacoTypes.languages.InlineCompletions<monacoTypes.languages.InlineCompletion>> {
+    const item: monacoTypes.languages.InlineCompletion = {
+      insertText: this.completion
+    };
+    console.log("provided docu completion")
```
Comment (Aniket-Engg) [inline libs/remix-ui/editor/src/lib/providers/documentationProvider.ts:17]: should this be removed?

File: libs/remix-ui/editor/src/lib/remix-ui-editor.tsx [hunk 3]
```diff
@@ -723,10 +733,46 @@ export const EditorUI = (props: EditorUIProps) => {
...
         const message = intl.formatMessage({ id: 'editor.generateDocumentationByAI' }, { content, currentFunction: currentFunction.current })
-        await props.plugin.call('solcoder', 'code_explaining', message)
+        const cm = await props.plugin.call('solcoder', 'code_explaining', message)
+
+        const natspecCom = "\n" + extractNatspecComments(cm)
+        const cln = await props.plugin.call('codeParser', "getLineColumnOfNode", currenFunctionNode)
+        const range = new monacoRef.current.Range(cln.start.line, cln.start.column, cln.start.line, cln.start.column)
+
+        const lines = natspecCom.split('\n')
+        const newnatspeccom = []
```
Comment (Aniket-Engg) [inline libs/remix-ui/editor/src/lib/remix-ui-editor.tsx:747]: can you update this var to camelCase?

**Review by Aniket-Engg — APPROVED** [review]

**Pull request merged**